docker/build-tools/prow-entrypoint.sh

```diff
@@ -35,4 +35,10 @@ if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
   gcloud auth configure-docker -q || true
 fi
 
-exec "$@"
+"$@"
+EXIT_VALUE=$?
+
+# Cleanup all docker artifacts
+docker ps -aq | xargs -r docker rm -f || true
```

Comment:
not currently relevant, but if this becomes more complex you might want this in a trap setup after starting docker, it might also be a good idea to stop the daemon.